### ISSUE TITLE
cargo build: unable to compile rcon 0.3.0

### ISSUE THREAD

**Issue by JensForstmann** [panicbit/rust-rcon#14]
After adding `rcon = "0.3.0"` to the `[dependencies]` section in `Cargo.toml` my project refuses to compile:

```
$ cargo build
    Blocking waiting for file lock on package cache
...

   Compiling rcon v0.3.0
error[E0599]: no method named `write_i32_le` found for struct `std::vec::Vec<_>` in the current scope
  --> C:\Users\Jens\.cargo\registry\src\github.com-1ecc6299db9ec823\rcon-0.3.0\src\packet.rs:71:13
   |
71 |         buf.write_i32_le(self.length).await?;
   |             ^^^^^^^^^^^^ method not found in `std::vec::Vec<_>`

error[E0599]: no method named `write_i32_le` found for struct `std::vec::Vec<_>` in the current scope
  --> C:\Users\Jens\.cargo\registry\src\github.com-1ecc6299db9ec823\rcon-0.3.0\src\packet.rs:72:13
   |
72 |         buf.write_i32_le(self.id).await?;
   |             ^^^^^^^^^^^^ method not found in `std::vec::Vec<_>`

error[E0599]: no method named `write_i32_le` found for struct `std::vec::Vec<_>` in the current scope
  --> C:\Users\Jens\.cargo\registry\src\github.com-1ecc6299db9ec823\rcon-0.3.0\src\packet.rs:73:13
   |
73 |         buf.write_i32_le(self.ptype.to_i32()).await?;
   |             ^^^^^^^^^^^^ method not found in `std::vec::Vec<_>`

error[E0599]: no method named `read_i32_le` found for mutable reference `&mut T` in the current scope
  --> C:\Users\Jens\.cargo\registry\src\github.com-1ecc6299db9ec823\rcon-0.3.0\src\packet.rs:83:24
   |
83 |         let length = r.read_i32_le().await?;
   |                        ^^^^^^^^^^^ method not found in `&mut T`

error[E0599]: no method named `read_i32_le` found for mutable reference `&mut T` in the current scope
  --> C:\Users\Jens\.cargo\registry\src\github.com-1ecc6299db9ec823\rcon-0.3.0\src\packet.rs:84:20
   |
84 |         let id = r.read_i32_le().await?;
   |                    ^^^^^^^^^^^ method not found in `&mut T`

error[E0599]: no method named `read_i32_le` found for mutable reference `&mut T` in the current scope
  --> C:\Users\Jens\.cargo\registry\src\github.com-1ecc6299db9ec823\rcon-0.3.0\src\packet.rs:85:23
   |
85 |         let ptype = r.read_i32_le().await?;
   |                       ^^^^^^^^^^^ method not found in `&mut T`

error: aborting due to 6 previous errors

For more information about this error, try `rustc --explain E0599`.
error: could not compile `rcon`.

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

```
$ rustc --version
rustc 1.46.0 (04488afe3 2020-08-24)

$ cargo --version
cargo 1.46.0 (149022b1d 2020-07-17)
```

**Comment by panicbit** [comment]
Odd, I'll look into it.

**Comment by panicbit** [comment]
I cannot replicate the issue (see image below), but maybe this is because `rcon` depends on stuff that was added in a patch version. Try running `cargo update` and then retry the build.
![image](https://user-images.githubusercontent.com/628445/93002037-0b9a7d80-f534-11ea-82e4-af1d2ecf01b8.png)

**Comment by panicbit** [comment]
Let me know if that helped, because then I need to bump the dep versions in `rcon`s `Cargo.toml`.

**Comment by panicbit** [comment]
Fixed in `0.3.1`. Just running `cargo update` should definitely fix the issue at this point.

**Comment by JensForstmann** [comment]
It works, **thanks**!
